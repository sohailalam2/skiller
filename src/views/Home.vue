<template>
  <div class="home">
    <h1 class="is-size-2 has-text-weight-bold is-hidden-mobile">ADD YOUR SKILLS</h1>

    <div class="columns">
      <SkillsSelector @addSkill="doAddSkills" class="column home__skills-selector"/>
    </div>

    <div class="columns is-multiline home__skills">
      <Skill v-for="(s, index) in skills"
             :id="s.id"
             :index="index + 1"
             :key="index"
             :name="s.name"
             :experience="s.experience"
             @deleteSkill="doDeleteSkill"
             class="column"/>
    </div>
  </div>
</template>

<script>
import { mapActions, mapState } from 'vuex';

import SkillsSelector from '../components/SkillsSelector.vue';
import Skill from '../components/Skill.vue';

export default {
  name: 'home',
  components: {
    SkillsSelector,
    Skill,
  },
  computed: {
    ...mapState(['skills']),
  },
  methods: {
    ...mapActions(['doAddSkills', 'doDeleteSkill']),
  },
};
</script>

<style lang="scss" scoped>
  .home {
    margin: 4rem;
    @media (max-width: 780px) {
      margin: 2rem;
    }

    &__skills-selector {
      margin-bottom: 1rem;
      @media (max-width: 780px) {
        padding: 0;
      }
    }

    &__skills {
      background: #F4F5F6;
    }
  }
</style>
