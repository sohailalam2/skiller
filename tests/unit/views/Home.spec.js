import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import Home from '@/views/Home.vue';

const localVue = createLocalVue();

localVue.use(Vuex);

describe('Home.vue', () => {
  let store;
  beforeEach(() => {
    store = new Vuex.Store({});
  });

  it('render home', () => {
    const wrapper = shallowMount(Home, { store, localVue });
    expect(wrapper.element).toMatchSnapshot();
  });
});
