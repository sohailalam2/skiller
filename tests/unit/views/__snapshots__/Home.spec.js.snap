// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home.vue render home 1`] = `
<div
  class="home"
>
  <h1
    class="is-size-2 has-text-weight-bold is-hidden-mobile"
  >
    ADD YOUR SKILLS
  </h1>
   
  <div
    class="columns"
  >
    <skillsselector-stub
      class="column home__skills-selector"
    />
  </div>
   
  <div
    class="columns is-multiline home__skills"
  />
</div>
`;
